src/modm/ui/color/rgb.hpp

```diff
@@ -0,0 +1,283 @@
+	 *
+	 * \param rgb565	RGB565 Color
+	 */
+	constexpr RgbT(const Rgb565& rgb565)
+		: red((rgb565.color >> 8) & 0xF8),
+		  green((rgb565.color >> 3) & 0xFC),
+		  blue(rgb565.color << 3)
+	{}
+
+	inline TSum
+	getColorSum() const
+	{
+		return red + green + blue;
+	}
+
+	template<modm::fundamental IntermediateType = float, unsigned int multiplier = 100,
+			 std::unsigned_integral ReturnType = T>
+	ReturnType
+	getRelative(const T color) const
+	{
+		return IntermediateType(color) * IntermediateType(multiplier) /
+			   IntermediateType(getColorSum());
+	}
+
+	/**
+	 * Normalize color values based on a clear value
+	 *
+	 * Imagine a low band light, for example a green laser. In case the filters
+	 * of a sensor do not transfer this wavelength well, it might result in all
+	 * colors being very low. An aditional clear value (provided f.e. by TCS3472)
+	 * will not filter colors and thus it will see a bright light (intensity).
+	 * In order to still have some signal the very low green value can be
+	 * amplified with the clear value.
+	 */
+	template<modm::fundamental IntermediateType = float>
+	void
+	normalizeByValue(T value)
+	{
+		const IntermediateType scalar = value / IntermediateType(getColorSum());
+		red *= scalar;
+		green *= scalar;
+		blue *= scalar;
+	}
+
+	template<modm::fundamental IntermediateType = float, unsigned int multiplier = 100,
+			 std::unsigned_integral ReturnType = T>
+	ReturnType
+	getRelativeRed() const
+	{
+		return getRelative<IntermediateType, multiplier, ReturnType>(red);
+	}
+
+	template<modm::fundamental IntermediateType = float, unsigned int multiplier = 100,
+			 std::unsigned_integral ReturnType = T>
+	ReturnType
+	getRelativeGreen() const
+	{
+		return getRelative<IntermediateType, multiplier, ReturnType>(green);
+	}
+
+	template<modm::fundamental IntermediateType = float, unsigned int multiplier = 100,
+			 std::unsigned_integral ReturnType = T>
+	ReturnType
+	getRelativeBlue() const
+	{
+		return getRelative<IntermediateType, multiplier, ReturnType>(blue);
+	}
+
+	template<modm::fundamental IntermediateType = float, unsigned int multiplier = 100,
+			 std::unsigned_integral ReturnType = T>
+	RgbT<ReturnType>
+	getRelativeColors() const
+	{
+		return RgbT<ReturnType>(getRelativeRed<IntermediateType, multiplier, ReturnType>(),
+								getRelativeGreen<IntermediateType, multiplier, ReturnType>(),
+								getRelativeBlue<IntermediateType, multiplier, ReturnType>());
+	}
+
+	constexpr bool
+	operator==(const RgbT<T>& other) const = default;
+
+	/**
+	 * \brief	Saturated addition ⊕
+	 */
+	RgbT<T>
+	operator+(const RgbT<T>& other) const
+	{
+		RgbT<T> result;
+		if (__builtin_add_overflow(red, other.red, &result.red))
+			result.red = numeric_limits<T>::max();
+		if (__builtin_add_overflow(green, other.green, &result.green))
+			result.green = numeric_limits<T>::max();
+		if (__builtin_add_overflow(red, other.blue, &result.blue))
+			result.blue = numeric_limits<T>::max();
+		return result;
+	}
```

Comment:
I also reimplement modm::Saturated. Here i quickly tested the modern solution using [Integer Overflow Builtins](https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html) for RgbT<T>::operator+.

**Works great + fast on all platforms!**